radareorg/cutter
# [WIP] Initial attempt at reworking renaming

**Your checklist for this pull request**


**Detailed description**



This PR aims at reworking the renaming workflow.
Basically the shortcut `N` for "renaming" should handle *every* case, such as:
- Renaming a function
- Renaming a local variable
- Renaming a flag
- Adding a flag

***WIP***

- [ ] Update `FlagDialog` so it can manage renaming/editing an existing flag
- [ ] Make sure the dialog code is consistent with `CutterSeekable` when it comes to select a location
- [ ] Make sure renaming a local variable works

**Test plan (required)**



For every case, it should be tried with a right click and with a direct call to the shortcut `N`.

- Rename a function
- Rename a function for which its first instruction contains a reference to an address e.g. `lea rdi, [0xc0fec0fe]`
- Rename an offset
- Rename a local variable
- Add a flag somewhere
- Delete a flag (make it empty)
- And so on...



**Closing issues**



This PR closes https://github.com/radareorg/cutter/issues/1698 and https://github.com/radareorg/cutter/issues/1211
